chain/sync.go

```diff
@@ -466,11 +466,18 @@ func (syncer *Syncer) ValidateTipSet(ctx context.Context, fts *store.FullTipSet)
 	}
 
 	for _, b := range fts.Blocks {
-		if err := syncer.ValidateBlock(ctx, b); err != nil {
-			if isPermanent(err) {
-				syncer.bad.Add(b.Cid(), err.Error())
+		isValidated, err := syncer.store.IsBlockValidated(ctx, b.Header)
```

Comment:
I think it probably makes sense to push this to `ValidateBlock`  (just check the cache before doing anything else in ValidateBlock, and return nil if found).

You can add the block to the cache in this method, though, after the call to `ValidateBlock`.

Reply:
fixed